src/types.ts

```diff
@@ -0,0 +1,52 @@
+interface ValidateTextOptionsBase {
+	medium?: 'all' | 'braille' | 'embossed' | 'handheld' | 'print' | 'projection' | 'screen' | 'speech' | 'tty' | 'tv';
+}
+
+type ValidateTextOptionsWithBaseWarning = ValidateTextOptionsBase & {
+	warningLevel?: 0;
+};
+
+type ValidateTextOptionsWithElevatedWarning = ValidateTextOptionsBase & {
+	warningLevel: 1 | 2 | 3;
+};
+
+export interface W3CCSSValidatorResponse {
+	cssvalidation: {
+		validity: boolean;
+		errors?: {
+			line: number;
+			message: string;
+		}[];
+		warnings?: {
+			line: number;
+			level: 1 | 2 | 3;
+			message: string;
+		}[];
+	};
+}
+
+export type ValidateTextOptions = ValidateTextOptionsWithElevatedWarning | ValidateTextOptionsWithBaseWarning;
+
+export interface ValidateTextResultBase {
+	valid: boolean;
+	errors: {
+		line: number;
+		message: string;
+	}[];
+}
+
+export type ValidateTextResultWithWarnings = ValidateTextResultBase & {
```

Comment:
Is there a reason why you used a `type` with the intersection operator (`&`), instead of `interface ValidateTextResultWithWarnings extends ValidateTextResultBase`?

Reply:
Not particularly, if we want to use extends I can change all the types to interfaces.

Reply:
Yeah, generally I think `extends` is a bit more clear, but it's not much of a difference.